tasks/ReedSwitchSimulatorTask.cpp

```diff
@@ -0,0 +1,147 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ReedSwitchSimulatorTask.hpp"
+#include <base-logging/Logging.hpp>
+
+using namespace std;
+using namespace base;
+using namespace gazebo_usv;
+using namespace linux_gpios;
+
+ReedSwitchSimulatorTask::ReedSwitchSimulatorTask(std::string const& name)
+    : ReedSwitchSimulatorTaskBase(name)
+{
+}
+
+ReedSwitchSimulatorTask::~ReedSwitchSimulatorTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See ReedSwitchSimulatorTask.hpp for more detailed
+// documentation about them.
+
+bool ReedSwitchSimulatorTask::validateGpio(linux_gpios::GPIOState const& gpio)
+{
+    if (gpio.states.size() != 1) {
+        LOG_FATAL_S << "received a gpio state of size " << gpio.states.size();
+        exception(INVALID_GPIO_SIZE);
+        return false;
+    }
+    return true;
+}
+
+bool ReedSwitchSimulatorTask::configureHook()
+{
+    if (!ReedSwitchSimulatorTaskBase::configureHook()) {
+        return false;
+    }
+    m_revolution_time = _revolution_time.get();
+    m_trigger_time = _trigger_time.get();
+    return true;
+}
+
+bool ReedSwitchSimulatorTask::startHook()
+{
+    if (!ReedSwitchSimulatorTaskBase::startHook()) {
+        return false;
+    }
+    m_timer = 0;
+    return true;
+}
+
+void ReedSwitchSimulatorTask::updateTimeInCycle(int64_t delta)
+{
+    m_timer += delta;
+    if (m_timer < 0) {
+        m_timer += m_revolution_time.microseconds;
+    }
+    m_timer = m_timer % m_revolution_time.microseconds;
```

Comment:
```suggestion
    auto updated_timer = m_timer + delta;
    if (updated_timer < 0) {
        updated_timer += m_revolution_time.microseconds;
    }
    m_timer = updated_timer % m_revolution_time.microseconds();
```

I would also add some comments to explain why you are using the timer like this.

Reply:
Also, I already spoke with Debora about this but for documentaiton purposes:

If delta is bigger than 1 revolution time and NEGATIVE, the flip on negative timer would NOT work. She will treat a delta with more than a revolution in the if clause